designdocs/capacity_staking_rewards_implementation.md

```diff
@@ -0,0 +1,324 @@
+1. **Staking Reward**: a per-RewardEra share of a staking reward pool of FRQCY tokens for a given staking account.
+1. **Reward Pool**:  a fixed amount of FRQCY that can be minted for rewards each RewardEra and distributed to stakers.
+1. **StakingRewardsProvider**: a trait that encapsulates the economic model for staking rewards, providing functionality for calculating the reward pool and staking rewards.
+
+## Staking Token Rewards
+
+### StakingAccountDetails updates
+New fields are added. The field `last_rewarded_at` is to keep track of the last time rewards were claimed for this Staking Account.
+MaximumCapacity staking accounts MUST always have the value `None` for `last_rewarded_at`.  This should be the default value also.
+`MaximumCapacity` is also the default value for `staking_type` and should map to 0.
+Finally, `stake_change_unlocking`, a BoundedVec is added which tracks the chunks of when a staking account has changed targets for some amount of funds.  This is to prevent retarget spamming.
+```rust
+pub struct StakingAccountDetails {
+    pub active: BalanceOf<T>,
+    pub total: BalanceOf<T>,
+    pub unlocking: BoundedVec<UnlockChunk<BalanceOf<T>, T::EpochNumber>, T::MaxUnlockingChunks>,
+    /// The number of the last StakingEra that this account's rewards were claimed.
+    pub last_rewards_claimed_at: Option<T::RewardEra>, // NEW  None means never rewarded, Some(RewardEra) means last rewarded RewardEra.
+    /// staking amounts that have been retargeted are prevented from being retargeted again for the
+    /// configured Thawing Period number of blocks.
+    pub stake_change_unlocking: BoundedVec<UnlockChunk<BalanceOf<T>, T::RewardEra>, T::MaxUnlockingChunks>, // NEW
+    /// total staked amounts for each past era, up to StakingRewardsPastErasMax eras.
+    pub staking_history: BoundedVec<StakingHistory<BalanceOf<T>, T::RewardEra>, T::StakingRewardsPastErasMax>, // NEW
+}
+
+/// A per-reward-era record for StakingAccount total_staked amount.
+pub struct StakingHistory<Balance, RewardEra> {
+    total_staked: Balance,
+    reward_era: RewardEra,
+}
+```
+
+### StakingTargetDetails updates
+A new field is added to indicate the type of staking the Account holder is doing in relation to this target.
+
+```rust
+#[derive(PartialEq, Eq, Clone, Encode, Decode, RuntimeDebug, TypeInfo, MaxEncodedLen)]
+#[scale_info(skip_type_params(T))]
+pub struct StakingTargetDetails<T: Config> {
+	/// The total amount of tokens that have been targeted to the MSA.
+	pub amount: BalanceOf<T>,
+	/// The total Capacity that an MSA received.
+	pub capacity: BalanceOf<T>,
+	/// The type of staking, which determines ultimate capacity per staked token.
+	pub staking_type: StakingType, // NEW
+}
+```
+
+**Unstaking thaw period**
+Changes the thaw period to begin at the first block of next RewardEra instead of immediately.
+
+### Changes to extrinsics
```

Comment:
I assume we are going to create new extrinsics instead of changing the existing ones but if we are not using them then changing might be fine

Reply:
Actually they are being used.  I'll update the document.